build-docs.sh

```diff
@@ -6,20 +6,8 @@
 set -e
 set -x
 
-python3 -V
-
-python3 -m pip install --user pdm
-
-python3 -m pdm install -G docs
-
-python3 -m pdm run python -c 'import docs.plugins.main'
-
-# Adding local symlinks gets nice source locations like
-#   pydantic_core/core_schema.py
-# instead of
-#   .venv/lib/python3.10/site-packages/pydantic_core/core_schema.py
-ln -s .venv/lib/python*/site-packages/pydantic_core pydantic_core
-ln -s .venv/lib/python*/site-packages/pydantic_settings pydantic_settings
-ln -s .venv/lib/python*/site-packages/pydantic_extra_types pydantic_extra_types
-
-python3 -m pdm run mkdocs build
+curl -LsSf https://astral.sh/uv/install.sh | sh
+${HOME}/.cargo/bin/uv python install 3.12
+${HOME}/.cargo/bin/uv sync --group docs --frozen
+${HOME}/.cargo/bin/uv run python -c 'import docs.plugins.main'
+${HOME}/.cargo/bin/uv run --no-sync mkdocs build
```

Comment:
If we added a shortcut to `Makefile` for this, should we use that here?

Reply:
if we have a makefile, do we need another shell script for this?

Reply:
Yes, this is still needed.